src/language/class-modifiers.md

```diff
@@ -0,0 +1,258 @@
+
+class Car extends          // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+    ...
+}
+
+class MockVehicle          // Can be implemented     
+  implements Vehicle {
+    @override
+    void moveForward ...
+}
+```
+
+To define a pure interface that others are free to implement, that cannot be
+inherited but is allowed to have abstract members, define an
+`abstract interface class`.
+
+## `final` 
+
+Use the `final` modifier to prevent subtyping from a class outside of the current
+library, closing the type hierarchy. 
+Disallowing both inheritance and implementation means `final` classes provide all
+the same guarantees of [`base`](#base) and [`interface`](#interface) classes.
+
+Final classes can be extended or implemented within the
+same library, but like the `base` modifier, any subtypes must be marked `base`, 
+`final`, or `sealed`.
+
+```dart
+// Library a.dart
+final class Vehicle {
+  void moveForward(int meters) { ... }
+}
+```
+
+```dart
+// Library b.dart
+import 'a.dart';
+
+var myCar = Vehicle();     // Can be constructed
+
+class Car extends          // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+    ...
+}
+
+class MockVehicle          // ERROR: Cannot be implemented     
+  implements Vehicle {
+    @override
+    void moveForward ...
+}
+```
+
+## `sealed`
+
+Use the `sealed` modifier to create a known, enumerable set of subtypes, so that
+a switch over those types can be statically ensured to be [_exhaustive_][exhaustive].
+
+The `sealed` modifier prevents a class from being extended or implemented outside
+of the library where the class is defined. The compiler is aware of any possible
+direct subtypes because they can only exist in the same library.
+
+This allows the compiler to alert you when a switch is not exhaustively handling
+all possible subtypes in its cases:
+
+```dart
+sealed class Vehicle { ... }
+
+class Car extends Vehicle { }
+class Truck implements Vehicle { }
+class Bicycle extends Vehicle { }
+
+// ...
+
+var vehicle = Vehicle();                 // ERROR: Cannot be instantiated
+
+// ...
+
+return switch (Vehicle vehicle) {       // ERROR: The switch is missing a subtype of Vehicle
+  Car => 'vroom',
+  Truck => 'VROOOOMM'
+};
+```
+
+If [exhaustiveness checking][exhaustive] is not necessary for your use case, use
+[`final`](#final) instead.
+
+## Combining modifiers
+
+It is possible to combine some modifiers for layered restrictions. 
+A class declaration can be, in order:
+
+* (Optional) `abstract`, describing whether the class can contain abstract members
+and prevents instantiation.
+* (Optional) One of `base`, `interface`, `final` or `sealed`, describing
+restrictions on other libraries subtyping the class.
+* (Optional) `mixin`, describing whether the declaration can be mixed in.
+* The `class` keyword itself.
+
+Some combinations are disallowed beacause they are contradictory, redundant, or
+otherwise mutually exclusive:
```

Comment:
**issues:** Write in active voice per [GSG](https://developers.google.com/style/voice). 
Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading). (Reading: 19.4 > 7.1)
**typo:** `because`, not `beacause`.
```suggestion
You can't combine some modifiers. They include the following:
```